### PR TITLE
Refactor LiveClient

### PR DESCRIPTION
**Pull request by dvonthenen**
Fixes: https://github.com/deepgram/deepgram-dotnet-sdk/issues/235
I am going to keep that issue open for right now in order to parallelize the firing of events, but this works! Will add interfaces to make things easier, but the existing ones are fairly close to stable.

Changes:
- Need to separate out the events into their own handlers. The issue is that the processing of each event blocks all other events. Since all events are funneled into a single event, depending on the amount of processing done on a handler, this could back the event queue up.

### CODE REVIEW

File: Deepgram/Clients/Live/v1/Client.cs [hunk 1]
```diff
@@ -44,10 +53,21 @@
...
     {
+        // check if the client is disposed
+        if (_clientWebSocket != null)
+        {
+            // client has already connected
+            // TODO: logging
+            var ex = new Exception("Client has already been initialized");
+            ProcessException("Connect", ex);
+            throw ex;
+        }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:64]: Will do in a subsequent PR.

File: Deepgram/Models/Live/v1/OpenResponse.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace Deepgram.Models.Live.v1;
+
+public record OpenResponse
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    [JsonPropertyName("type")]
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public LiveType? Type { get; set; } = LiveType.Open;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Live/v1/OpenResponse.cs:14]: Will do in a subsequent PR.

File: Deepgram/Models/Live/v1/UnhandledResponse.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    /// </summary>
+    [JsonPropertyName("raw")]
+    public string? Raw { get; set; } = "";
+
+    /// <summary>
+    /// TODO
+    /// </summary>
+    [JsonPropertyName("type")]
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public LiveType? Type { get; set; } = LiveType.Unhandled;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Live/v1/UnhandledResponse.cs:20]: Will do in a subsequent PR.

File: Deepgram/Models/Live/v1/ErrorResponse.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    /// </summary>
+    [JsonPropertyName("variant")]
+    public string? Variant { get; set; } = "";
+
+    /// <summary>
+    /// TODO
+    /// </summary>
+    [JsonPropertyName("type")]
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public LiveType? Type { get; set; } = LiveType.Error;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Live/v1/ErrorResponse.cs:32]: Will do in a subsequent PR.

File: Deepgram/Models/Live/v1/CloseResponse.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+namespace Deepgram.Models.Live.v1;
+
+public record CloseResponse
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    [JsonPropertyName("type")]
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public LiveType? Type { get; set; } = LiveType.Close;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Live/v1/CloseResponse.cs:14]: Will do in a subsequent PR.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 6]
```diff
@@ -195,26 +256,96 @@
...
-                    {
+                    {         
+                        case LiveType.Open:
+                            //var openResponse = new ResponseEvent<OpenResponse>(data.Deserialize<OpenResponse>());
+                            var openResponse = data.Deserialize<OpenResponse>();
+                            if (_openReceived == null || openResponse == null)
+                            {
+                                return;
+                            }
+                            _openReceived.Invoke(null, data.Deserialize<OpenResponse>());
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:272]: Will do in a subsequent PR.

**Review by SandraRodgers — APPROVED** [review]

**Pull request merged**